test/e2e/es/mutation_test.go

```diff
@@ -160,6 +160,25 @@ func TestMutationSecondMasterSetDown(t *testing.T) {
 	RunESMutation(t, b, mutated)
 }
 
+// TestMutationRollingDownscaleCombination combines a rolling update with scale down operation.
+func TestMutationRollingDownscaleCombination(t *testing.T) {
```

Comment:
Makes me realize we probably also want an E2E test to combine rolling and downscale of the same nodeSet (outside the scope of this PR).